playback/src/audio_backend/alsa.rs

```diff
@@ -1,14 +1,81 @@
 use super::{Open, Sink};
-use alsa::{Access, Format, Mode, Stream, PCM};
+use alsa::device_name::HintIter;
+use alsa::pcm::{Access, Format, HwParams, PCM};
+use alsa::{Direction, Error, ValueOr};
+use std::env;
+use std::ffi::CString;
 use std::io;
+use std::process::exit;
 
 pub struct AlsaSink(Option<PCM>, String);
 
+fn list_outputs() {
+    for t in &["pcm", "ctl", "rawmidi", "timer", "seq", "hwdep"] {
+        println!("{} devices:", t);
+        let i = HintIter::new(None, &*CString::new(*t).unwrap()).unwrap();
+        for a in i {
+            println!("  {:?}", a)
+        }
+    }
+}
+
+fn open_device(dev_name: &str) -> Result<(PCM), Box<Error>> {
+    let pcm = PCM::new(dev_name, Direction::Playback, false)?;
+    // http://www.linuxjournal.com/article/6735?page=0,1#N0x19ab2890.0x19ba78d8
+    // latency = period_size * periods / (rate * bytes_per_frame)
+    // For 16 Bit stereo data, one frame has a length of four bytes.
+    // 500ms  = buffer_size / (44100 * 4)
+    // buffer_size = 0.5 * 44100 = 22050 frames
```

Comment:
Why this buffer size?

Reply:
The buffer size is derived from the latency of `500000 us` originally set by @plietar in [`rust-alsa`](https://github.com/plietar/rust-alsa/blob/8c63543fa0ccd971cf15f5675293d19febd6f79e/src/alsa.rs#L115)

Reply:
Ah, thought it was clear it was being expressed in frames, but sure can make it clearer!